scripts/token-paymaster-v2/deploy-token-paymaster.ts

```diff
@@ -0,0 +1,128 @@
+import { ethers, run } from "hardhat";
+import {
+  deployContract,
+  DEPLOYMENT_SALTS,
+  encodeParam,
+  isContract,
+} from "../utils";
+import { Deployer, Deployer__factory } from "../../typechain-types";
+
+const provider = ethers.provider;
+const entryPointAddress =
+  process.env.ENTRY_POINT_ADDRESS ||
+  "0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789";
+const verifyingSigner = process.env.PAYMASTER_SIGNER_ADDRESS_PROD || "";
+const DEPLOYER_CONTRACT_ADDRESS =
+  process.env.DEPLOYER_CONTRACT_ADDRESS_PROD || "";
+
+function delay(ms: number) {
+  return new Promise<void>((resolve) => {
+    setTimeout(() => {
+      resolve();
+    }, ms);
+  });
+}
+
+async function deployTokenPaymasterContract(
+  deployerInstance: Deployer,
+  earlyOwnerAddress: string
+): Promise<string | undefined> {
+  try {
+    const salt = ethers.utils.keccak256(
+      ethers.utils.toUtf8Bytes(DEPLOYMENT_SALTS.TOKEN_PAYMASTER_V2)
+    );
```

Comment:
It would be better to create a common `deployGeneric` similar to https://github.com/bcnmy/scw-contracts/blob/v2-deployments-with-config/scripts/deploy.ts#L67 and use it across all scripts instead of duplicating deploy -> verify logic

Reply:
added deployGeneric() in this script and can re use for other contracts in same script (although there aren't any now)